tests/testthat/test-image.R

```diff
@@ -1,76 +1,125 @@
-  #     fn = function(x) {
-  #       x <- as.numeric(x)
-  #       ifelse(
-  #         x >= 5,
-  #         local_image(file = test_image(type = "svg"), height = px(x * 10)),
-  #         local_image(file = test_image(type = "png"), height = px(x * 10)))
-  #     })
-  #
-  # # Expect that PNG image tags are in the rendered table
-  # tbl_html %>%
-  #   render_as_html() %>%
-  #   expect_match("<img cid=.*?test_image.png. src=.data:image.png;base64.*?")
-  #
-  # # Expect that SVG image tags are in the rendered table
-  # tbl_html %>%
-  #   render_as_html() %>%
-  #   expect_match("<img cid=.*?test_image.svg. src=.data:image.svg.xml;base64.*?")
+  testthat::local_edition(3)
+
+  # Expect that the image tags generated for the included test image
+  # are correctly formed
+  expect_snapshot(local_image(filename = test_image(type = "png")))
+  expect_snapshot(local_image(filename = test_image(type = "svg")))
+
+  # Expect that `local_image()` can return multiple image tags
+  expect_snapshot(local_image(filename = rep(test_image(type = "png"), 2)))
 })
 
 test_that("the `web_image()` function works correctly", {
 
-  url <- "https://upload.wikimedia.org/wikipedia/commons/thumb/3/30/Googlelogo.png/320px-Googlelogo.png"
+  # Expect that the image tag is correctly formed
+  expect_equal(
+    web_image(url = "example.png"),
+    "<img src=\"example.png\" style=\"height:30px;\">"
+  )
 
   # Expect that the `web_image()` function returns an image tag
-  web_image(url = url, height = px(200)) %>%
-    expect_match(paste0("^<img src=.", url, ". style=.height:200px;.>"))
+  expect_equal(
+    web_image(url = "example.com/img.png", height = px(200)),
+    "<img src=\"example.com/img.png\" style=\"height:200px;\">"
+  )
 
-  # Expect that the `web_image()` function returns an image tag
-  # (even if a number is provided to `height`)
-  web_image(url = url, height = 200) %>%
-    expect_match(paste0("^<img src=.", url, ". style=.height:200px;.>"))
+  # Expect that the `web_image()` function returns the same image tag
+  # even if `height` is provided in different ways
+  expect_equal(
+    web_image(url = "example.com/img.png", height = px(200)),
+    web_image(url = "example.com/img.png", height = 200)
+  )
 
-  # Expect that the `web_image()` function returns an image tag
-  # (even if a string is provided to `height`)
-  web_image(url = url, height = "200px") %>%
-    expect_match(paste0("^<img src=.", url, ". style=.height:200px;.>"))
+  expect_equal(
+    web_image(url = "example.com/img.png", height = 200),
+    web_image(url = "example.com/img.png", height = "200px")
+  )
+
+  # Expect that `web_image()` can return multiple image tags
+  expect_equal(
+    web_image(
+      url = c("example.com/img1.png", "example.com/img2.png"),
+      height = px(200)
+    ),
+    paste0("<img src=\"example.com/img", 1:2, ".png\" style=\"height:200px;\">")
+  )
+})
+
+test_that("the `ggplot_image()` function works correctly", {
+
```

Comment:
Can we just skip on CI here, not at the top of this file?

Reply:
Now done.